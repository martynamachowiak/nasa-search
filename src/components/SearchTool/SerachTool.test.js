import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { SearchTool } from "../";

describe("SearchTool", () => {
  it("renders expected tree", () => {
    const tree = renderer.create(<SearchTool onSubmit={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("submits search query on button click", () => {
    const onSubmit = jest.fn();
    render(<SearchTool onSubmit={() => onSubmit()} />);
    fireEvent.click(screen.getByRole("button"));
    expect(onSubmit).toBeCalledTimes(1);
  });

  describe("selects correctly", () => {
    const cases = ["Images", "Videos"];

    it.each(cases)("checkbox with label %s when clicked", (label) => {
      const { getByLabelText } = render(<SearchTool onSubmit={() => {}} />);
      const checkboxImages = getByLabelText(label);
      fireEvent.click(checkboxImages);
      expect(checkboxImages).toBeChecked();
    });
  });
});
