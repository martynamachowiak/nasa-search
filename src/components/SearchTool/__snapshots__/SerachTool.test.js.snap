// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchTool renders expected tree 1`] = `
<div
  className="search-tool"
>
  <div
    className="search-tool__query"
  >
    <input
      onChange={[Function]}
      placeholder="e.g. space cat"
      type="text"
      value=""
    />
    <button
      onClick={[Function]}
    >
      Search
    </button>
  </div>
  <div
    className="search-tool__criteria"
  >
    <label>
      <input
        name="image"
        onChange={[Function]}
        type="checkbox"
      />
      Images
    </label>
    <label>
      <input
        name="video"
        onChange={[Function]}
        type="checkbox"
      />
      Videos
    </label>
  </div>
</div>
`;
